### PR TITLE
Add telemetry properties for NuGet restore cycle detector.

### PR DESCRIPTION
**Pull request by ocallesp**
Implements #8389 

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/8403)

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/PackageRestoreDataSource.cs [hunk 1]
```diff
@@ -208,6 +217,18 @@ private async Task<bool> CycleDetectedAsync(byte[] hash, CancellationToken cance
             );
 
             return true;
+
+            void SendTelemetry()
+            {
+                _stopwatch.Stop();
```
[the remark this answers is not in the document]
Comment (MiYanni) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/PackageRestoreDataSource.cs:223]: This is a local function within CycleDetectedAsync method. Not sure how local functions work behind-the-scenes, though.

[the remark this answers is not in the document]
Comment (ocallesp) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/PackageRestoreDataSource.cs]: Done. CycleDetectedAsync () now starts and stops the stopWatch

**Review by jacdavis — APPROVED** [review]
:shipit:

**Comment by ocallesp** [comment]
git squash

**Pull request merged**